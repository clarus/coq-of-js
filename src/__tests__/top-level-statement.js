// @flow
import {compileAndPrint} from "../compiler/index.js";

it("does not handle destructuring of constants", () => {
  expect(compileAndPrint(`const [a, b] = [1, 2];`)).toMatchSnapshot();
});

it("handles React import", () => {
  expect(
    compileAndPrint(`import React, {PureComponent} from "react";`),
  ).toMatchSnapshot();
});

it("does not handle non-React imports", () => {
  expect(compileAndPrint(`import express from "express";`)).toMatchSnapshot();
});

it("does not handle empty declarations", () => {
  expect(compileAndPrint(`let x;`)).toMatchSnapshot();
});
