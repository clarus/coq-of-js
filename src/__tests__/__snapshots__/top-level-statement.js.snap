// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`does not handle destructuring of constants 1`] = `
"> 1 | const [a, b] = [1, 2];
    |      ^^^^^^

Expected simple identifier"
`;

exports[`does not handle empty declarations 1`] = `
"> 1 | let x;
    |    ^

Expected definition"
`;

exports[`does not handle non-React imports 1`] = `
"> 1 | import express from \\"express\\";
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Only handle imports from React for now"
`;

exports[`handles React import 1`] = `""`;
